### PR TITLE
Ins 877 networkcoordinator unittests

### PR DESCRIPTION
**Pull request by ilyapt**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: networkcoordinator/networkcoordinator_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/insolar/insolar/core/reply"
+	"github.com/insolar/insolar/log"
+	"github.com/insolar/insolar/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewNetworkCoordinator(t *testing.T) {
+	nc, err := New()
+	require.NoError(t, err)
+	require.Equal(t, &NetworkCoordinator{}, nc)
```
Comment (megge-dream) [inline networkcoordinator/networkcoordinator_test.go]: You can add check if injected component was really inject - use mocks, 
`&component.Manager{}.Inject(mockComponent1, mockComponent2, ..., nc)`
and check if they was setted - 
`require.Equal(t, nc.CertificateManager, mockCertificateManager)`

File: networkcoordinator/networkcoordinator_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	nc, err := New()
+	require.NoError(t, err)
+	mb := testutils.NewMessageBusMock(t)
+	mb.MustRegisterFunc = func(p core.MessageType, handler core.MessageHandler) {
+		require.Equal(t, p, core.TypeNodeSignRequest)
+	}
+	nc.MessageBus = mb
+	ctx := context.Background()
+	err = nc.Start(ctx)
+	require.NoError(t, err)
```
Comment (megge-dream) [inline networkcoordinator/networkcoordinator_test.go]: Check that `zeroCoordinator` and `realCoordinator` not empty, and `isStarted` is true

File: networkcoordinator/networkcoordinator_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+func TestNetworkCoordinator_GetCoordinator_Zero(t *testing.T) {
+	nc, err := New()
+	require.NoError(t, err)
+	ns := testutils.NewNetworkSwitcherMock(t)
+	ns.GetStateFunc = func() core.NetworkState {
+		return core.NoNetworkState
+	}
+	nc.NetworkSwitcher = ns
+	crd := nc.getCoordinator()
+	require.Equal(t, nc.zeroCoordinator, crd)
```
Comment (megge-dream) [inline networkcoordinator/networkcoordinator_test.go]: Is `nc.zeroCoordinator` not nil here? I don't see it setting above.

File: networkcoordinator/networkcoordinator_test.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+func TestNetworkCoordinator_GetCoordinator_Real(t *testing.T) {
+	nc, err := New()
+	require.NoError(t, err)
+	ns := testutils.NewNetworkSwitcherMock(t)
+	ns.GetStateFunc = func() core.NetworkState {
+		return core.CompleteNetworkState
+	}
+	nc.NetworkSwitcher = ns
+	crd := nc.getCoordinator()
+	require.Equal(t, nc.realCoordinator, crd)
```
Comment (megge-dream) [inline networkcoordinator/networkcoordinator_test.go]: Is `nc.realCoordinator` not nil here? I don't see it setting above.

File: networkcoordinator/real_test.go [hunk 5]
```diff
@@ -0,0 +1,837 @@
...
+	_, err := coord.GetCert(ctx, &nodeRef)
+	require.EqualError(t, err, "[ GetCert ] Couldn't create certificate: test_error")
+}
+
+func TestRealNetworkCoordinator_GetCert_SignCertError(t *testing.T) {
+	nodeRef := testutils.RandomRef()
+	certNodeRef := testutils.RandomRef().String()
+
+	cr := testutils.NewContractRequesterMock(t)
+	cr.SendRequestFunc = func(ctx context.Context, ref *core.RecordRef, method string, args []interface{}) (core.Reply, error) {
```
Comment (LinnikD) [inline networkcoordinator/real_test.go]: You can init this func inside NewContractRequesterMock func

File: networkcoordinator/networkcoordinator_test.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+func test_getNode() *reply.CallMethod {
```
Comment (megge-dream) [inline networkcoordinator/networkcoordinator_test.go]: Maybe name it `mockReply` ?

File: networkcoordinator/networkcoordinator_test.go [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+	node, err := core.MarshalArgs(struct {
+		PublicKey string
+		Role      core.StaticRole
+	}{
+		PublicKey: "test_node_public_key",
+		Role:      core.StaticRoleVirtual,
+	}, nil)
+	if err != nil {
+		log.Fatal(err)
```
Comment (megge-dream) [inline networkcoordinator/networkcoordinator_test.go]: Better use here `require.NoError(t, err)`

**Pull request merged**